Update OIDC client documentation with examples and clarifications (#683)
DuendeSoftware/docs.duendesoftware.com: src/content/docs/identitymodel-oidcclient/automatic.md

  - /foss/identitymodel.oidcclient/automatic/
---

> OpenID Connect (OIDC) is a simple identity layer on top of the OAuth 2.0

Reviewer: ```suggestion
OpenID Connect (OIDC) is an identity layer on top of the OAuth 2.0
```

Not _that_ simple in many cases :-)

    /// </summary>
    /// <param name="options">The options.</param>
    /// <param name="cancellationToken">A token that can be used to cancel the request</param>
    /// <returns></returns>
    Task<BrowserResult> InvokeAsync(BrowserOptions options, CancellationToken cancellationToken = default);
}
```

The `BrowserResult` represents the result of the browser interaction, including any OIDC payloads that
are returned from the authentication server.

```csharp
// Copyright (c) Duende Software. All rights reserved.
// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.

namespace Duende.IdentityModel.OidcClient.Browser;

/// <summary>
/// The result from a browser login.
/// </summary>
/// <seealso cref="Result" />
public class BrowserResult : Result
{
    /// <summary>
    /// Gets or sets the type of the result.
    /// </summary>
    /// <value>
    /// The type of the result.
    /// </value>
    public BrowserResultType ResultType { get; set; }

    /// <summary>
    /// Gets or sets the response.
    /// </summary>
    /// <value>
    /// The response.
    /// </value>
    public string Response { get; set; }
}
```

> :::note
> The `IBrowser` implementation must be specific to the platform and environment and must be provided by the
> host application. For example, a Windows-specific implementation will not work within a macOS, iOS, Android, or Linux environment.
> :::

Reviewer: ```suggestion
:::note[Browser is platform-specific]
The `IBrowser` implementation is specific to the platform and environment and must be provided by the
host application. For example, a Windows-specific implementation will not work within a macOS, iOS, Android, or Linux environment.
:::
```

For a simple example, the following code shows how to use the
[SystemBrowser](https://github.com/DuendeSoftware/foss/blob/main/identity-model-oidc-client/clients/ConsoleClientWithBrowser/SystemBrowser.cs)
to invoke a browser on the host desktop platform. The `SystemBrowser` is a naive implementation that uses the

> [System.Diagnostics.Process](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process) class to invoke the browser.

Reviewer: ```suggestion
[System.Diagnostics.Process](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process) class to start the system default browser.
```
